app/(chat)/api/chat/route.ts

```diff
@@ -69,11 +70,13 @@ export async function POST(request: Request) {
     });
   }
 
+  const geoLocation = geolocation(request);
+
   return createDataStreamResponse({
     execute: (dataStream) => {
       const result = streamText({
         model: myProvider.languageModel(selectedChatModelId),
-        system: systemPrompt({ selectedChatModelId }),
+        system: systemPrompt({ selectedChatModelId, geoLocation }),
```

Comment:
That's awesome 👏 

Along the same lines, here's another idea:

Shall we also include the current date & time from the user and inject it into the system prompt?

This would need to happen client-side. Hence, we would need to use `experimental_prepareRequestBody` from the AI SDK in the `chat.tsx` component and could send an ISO 8601 formatted date to the Chat endpoint.

I can raise a PR for it if you think it's a good idea?

Reply:
Hey @tobiasbueschel, thanks for leaving feedback!

I'm currently testing this on the `live` branch to test its stability before I add it to the main template. Regardless, this is a step forward towards personalizing the system prompt based on dynamic values like location and time, so feel free to open a PR to include the current date & time.

I wonder if it would be necessary to pass that data from the client and instead can use the geolocation data or the other values returned from the [geolocation api](https://vercel.com/docs/functions/functions-api-reference/vercel-functions-package) to ascertain the timezone and convert from UTC to local time.